### DISCUSSION TITLE
how to insert body

### DISCUSSION THREAD

**Discussion by thisoban** [kamranayub/igdb-dotnet#19] [Q&A] [answered]
public IGDBClient Client { get; }

        public IGDBService()
        {
            Client = new IGDBClient("token", "secret");
        }

        public async Task<Game> GetGame(int id)
        {
            var games = await Client.QueryAsync<Game>(IGDBClient.Endpoints.Games, query: $"fields id,name; where id ={id} ;");
            var game = games.First();

            Console.WriteLine(game.Name, game.Id);
            return game;
        }

i am trying to do this but i dont understand where is the body and how do i give it as well

**Comment by kamranayub** [comment, marked as answer]
Hey there. Not sure what you mean by "body"? When you try to run this query, are you running into an exception or is it working?

**Reply by thisoban** [reply]
its working but i need to know how to send a body like i want specific stuff from  i wanne fill this in  how do i do that? fields *; where id = 1942;

**Reply by kamranayub** [reply]
It takes the query expression as a string just like you have it. The way I do it I tweak it first using the REST API but the documentation for how to build a query and the expression syntax is on the IGDB docs:

https://api-docs.igdb.com/#apicalypse-1

You could build the query string any way you want, like using a `StringBuilder`. That's probably the best way to handle it currently.

There's no strongly typed way to do this right now.
